src/Modules/SimplCommerce.Module.StorageLocal/LocalStorageService.cs

```
@@ -16,19 +16,29 @@ public string GetMediaUrl(string fileName)
 
         public async Task SaveMediaAsync(Stream mediaBinaryStream, string fileName, string mimeType = null)
         {
-            var filePath = Path.Combine(GlobalConfiguration.WebRootPath, MediaRootFoler, fileName);
+            var dirPath = Path.Combine(GlobalConfiguration.WebRootPath, MediaRootFoler);            
+            if (!Directory.Exists(dirPath))
+            {
+                Directory.CreateDirectory(dirPath);
+            }
+            var filePath = Path.Combine(dirPath, fileName);
             using (var output = new FileStream(filePath, FileMode.Create))
             {
                 await mediaBinaryStream.CopyToAsync(output);
+               
             }
         }
 
         public async Task DeleteMediaAsync(string fileName)
         {
-            var filePath = Path.Combine(GlobalConfiguration.WebRootPath, MediaRootFoler, fileName);
```

I prefer to throw an exception here instead

---

yes you can

---

Please do it and let us merge this